xonsh/foreign_shells.py

```diff
@@ -173,6 +177,11 @@ def foreign_shell_data(shell, interactive=True, login=False, envcmd=None,
     funcscmd = DEFAULT_FUNCSCMDS.get(shell, 'echo {}') if funcscmd is None else funcscmd
     command = COMMAND.format(envcmd=envcmd, aliascmd=aliascmd, prevcmd=prevcmd,
                              postcmd=postcmd, funcscmd=funcscmd).strip()
+    if shell == 'cmd':
+        cmd.remove('-c')
+        cmd.append('/C')
+        command = command.replace('\n','&&')
+
```

Comment:
The `-c` or `/C` thing is something that should be looked up in a top-level dictionary (like `DEFAULT_ENVCMDS`), or perhaps be a kwarg to this function.  That will fix the remove and append lines.